src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -1,311 +1,291 @@
+        assertTrue(service.isRunning());
+    }
+
+    @Test public void checkAbilityToStartServiceUsingFlags() throws Exception {
+        service = new AppiumServiceBuilder()
+            .withArgument(CALLBACK_ADDRESS, testIP)
+            .withArgument(SESSION_OVERRIDE)
+            .withArgument(PRE_LAUNCH)
+            .build();
+        service.start();
+        assertTrue(service.isRunning());
+    }
+
+    @Test public void checkAbilityToStartServiceUsingCapabilities() throws Exception {
+        File appDir = new File("src/test/java/io/appium/java_client");
+        File app = new File(appDir, "ApiDemos-debug.apk");
+        File pageFactoryDir = new File("src/test/java/io/appium/java_client/pagefactory_tests");
+        File chrome = new File(pageFactoryDir, "chromedriver.exe");
+
+        DesiredCapabilities caps = new DesiredCapabilities();
+        caps.setCapability(PLATFORM_NAME, "Android");
+        caps.setCapability(FULL_RESET, true);
+        caps.setCapability(NEW_COMMAND_TIMEOUT, 60);
+        caps.setCapability(APP_PACKAGE, "io.appium.android.apis");
+        caps.setCapability(APP_ACTIVITY, ".view.WebView1");
+        caps.setCapability(APP, app.getAbsolutePath());
+        caps.setCapability(CHROMEDRIVER_EXECUTABLE, chrome.getAbsolutePath());
+
+        service = new AppiumServiceBuilder().withCapabilities(caps).build();
+        service.start();
+        assertTrue(service.isRunning());
+    }
+
+    @Test public void checkAbilityToStartServiceUsingCapabilitiesAndFlags() throws Exception {
+        File appDir = new File("src/test/java/io/appium/java_client");
+        File app = new File(appDir, "ApiDemos-debug.apk");
+        File pageFactoryDir = new File("src/test/java/io/appium/java_client/pagefactory_tests");
+        File chrome = new File(pageFactoryDir, "chromedriver.exe");
+
+        DesiredCapabilities caps = new DesiredCapabilities();
+        caps.setCapability(PLATFORM_NAME, "Android");
+        caps.setCapability(FULL_RESET, true);
+        caps.setCapability(NEW_COMMAND_TIMEOUT, 60);
+        caps.setCapability(APP_PACKAGE, "io.appium.android.apis");
+        caps.setCapability(APP_ACTIVITY, ".view.WebView1");
+        caps.setCapability(APP, app.getAbsolutePath());
+        caps.setCapability(CHROMEDRIVER_EXECUTABLE, chrome.getAbsolutePath());
+
+        service = new AppiumServiceBuilder()
+                .withArgument(CALLBACK_ADDRESS, testIP)
+                .withArgument(SESSION_OVERRIDE)
+                .withArgument(PRE_LAUNCH)
+                .withCapabilities(caps).build();
+        service.start();
+        assertTrue(service.isRunning());
+    }
+
+    @Test public void checkAbilityToChangeOutputStream() throws Exception {
+        file = new File("test");
+        file.createNewFile();
+        stream = new FileOutputStream(file);
+        service = buildDefaultService();
+        service.addOutPutStream(stream);
+        service.start();
+        assertTrue(file.length() > 0);
```

Comment:
-> assertThat(file.length(), greaterThan(0L))

Reply:
this is a bit longer, but produces better error message in comparison to the generic True assertion